presto-product-tests/src/main/java/io/prestosql/tests/hive/TestIgnoreAbsentPartition.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.hive;
+
+import com.google.inject.Inject;
+import io.prestosql.tempto.ProductTest;
+import io.prestosql.tempto.Requirement;
+import io.prestosql.tempto.RequirementsProvider;
+import io.prestosql.tempto.configuration.Configuration;
+import io.prestosql.tempto.fulfillment.table.MutableTablesState;
+import io.prestosql.tempto.hadoop.hdfs.HdfsClient;
+import org.testng.annotations.Test;
+
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tempto.fulfillment.table.MutableTableRequirement.State.LOADED;
+import static io.prestosql.tempto.fulfillment.table.TableRequirements.mutableTable;
+import static io.prestosql.tempto.query.QueryExecutor.query;
+import static io.prestosql.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.prestosql.tests.hive.HiveTableDefinitions.NATION_PARTITIONED_BY_BIGINT_REGIONKEY;
+import static java.lang.String.format;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestIgnoreAbsentPartition
+        extends ProductTest
+        implements RequirementsProvider
+{
+    @Inject
+    private MutableTablesState tablesState;
+
+    @Inject
+    private HdfsClient hdfsClient;
+
+    @Override
+    public Requirement getRequirements(Configuration configuration)
+    {
+        return mutableTable(NATION_PARTITIONED_BY_BIGINT_REGIONKEY, "test_table", LOADED);
+    }
+
+    /**
+     * This test requires hive.ignore-absent-partitions=true
+     */
+    @Test(groups = PROFILE_SPECIFIC_TESTS)
+    public void testIgnoreAbsentPartitions()
+    {
+        String tableNameInDatabase = tablesState.get("test_table").getNameInDatabase();
+
+        assertThat(query("SELECT count(*) FROM " + tableNameInDatabase)).containsOnly(row(15));
+
+        String partitionPath = format("/user/hive/warehouse/%s/p_regionkey=9999", tableNameInDatabase);
+        assertFalse(hdfsClient.exist(partitionPath));
+        query(format("CALL hive.system.create_empty_partition('default', '%s', array['p_regionkey'], array['9999'])", tableNameInDatabase));
+        assertTrue(hdfsClient.exist(partitionPath));
+        hdfsClient.delete(partitionPath);
+        assertFalse(hdfsClient.exist(partitionPath));
```

Comment:
please add a message to assertFalse/True, otherwise the raised exc msg will be little hard to use